File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int max_sum = 0;
        int curr_sum = 0;
        for(int i =0; i<k;i++) {
            curr_sum+=nums[i];
            max_sum = curr_sum;
        }

        int left = 0, right = k;
        while(right<nums.length) {
            curr_sum += nums[right++];
            curr_sum -= nums[left++];
            max_sum = Math.max(max_sum, curr_sum);
        }
        System.out.println("max_sum: " + max_sum);
        return (double) max_sum / k;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        maxSum = currSum = sum(nums[:k])
        l, r = 1, k
        while l <= r and r < len(nums):
            currSum -= nums[l-1]
            currSum+= nums[r]
            maxSum = max(maxSum, currSum) 
            l+=1
            r+=1
        
        return maxSum / k